torch/csrc/jit/docs/OVERVIEW.md

```diff
@@ -548,7 +549,97 @@ This makes it possible to use most of the compiler functionality when python is
 
 [script/compiler.cpp](../script/compiler.cpp)
 
-The Environment object tracks the assignment of variable names to SugaredValues during compilation. It is local to the compiler file. A stack of environments exist, with a new environment being created for sub-blocks introduced by control flow. The Environment also handles turning the AST representation into SSA-form by tracking which variables were modified inside a sub-block and inserting the correct inputs/outputs to the Blocks of if-statements and loops.
+The Environment object tracks the assignment of variable names during compilation. It is local to the compiler file. A stack of environments exist, with a new environment being created for sub-blocks introduced by control flow. The Environment keeps two tables, one for values which are not first class in the type system (Sugared values) and a type table for values which are. When first class values are set, we emit a prim::Store, and when they are referenced we emit a prim::Load. Sugared values are not re-assignable. The graph is converted to SSA in the convertToSSA pass.
+
+## Conversion To SSA ##
+
+[script/convert_to_ssa.cpp](../script/convert_to_ssa.cpp)
+
+Converting to SSA works in multiple parts.
```

Comment:
Could you please add a high level overview of why we're doing this (in contrast to "what")? It might be not obvious why this all is necessary and how it relates to classic SSA. For instance, in the conventional definition of SSA the central part is phi-nodes. We don't have them because we work only with structured control flow, but we never mentioned that here. But that's what explains all of what's happening here: we're trying to represent constructs with breaks/continues/returns in a form of well-structured CFG that only have loops with no early exits and ifs.
Another thing that might be confusing is the load/store part - we're not loading or storing anything there, it's just a trick we use to figure out input/output values of a block, and I think it's also worth mentioning somewhere.